packages/flutter/lib/src/widgets/display_feature_sub_screen.dart

```diff
@@ -0,0 +1,233 @@
+      } else if (point.dy > rect.bottom) {
+        // Case 6
+        return (point - rect.bottomLeft).distance;
+      } else {
+        // Case 4
+        return rect.left - point.dx;
+      }
+    } else if (point.dx > rect.right) {
+      if (point.dy < rect.top) {
+        // Case 3
+        return (point - rect.topRight).distance;
+      } else if (point.dy > rect.bottom) {
+        // Case 8
+        return (point - rect.bottomRight).distance;
+      } else {
+        // Case 5
+        return point.dx - rect.right;
+      }
+    } else {
+      if (point.dy < rect.top) {
+        // Case 2
+        return rect.top - point.dy;
+      } else if (point.dy > rect.bottom) {
+        // Case 7
+        return point.dy - rect.bottom;
+      } else {
+        // Case R
+        return 0;
+      }
+    }
+  }
+
+  /// Returns sub-screens resulted by dividing [wantedBounds] along items of
+  /// [avoidBounds] that are at least as high or as wide.
+  static Iterable<Rect> _subScreensInBounds(Rect wantedBounds, Iterable<Rect> avoidBounds) {
+    Iterable<Rect> subScreens = <Rect>[wantedBounds];
+    for (final Rect bounds in avoidBounds) {
+      subScreens = subScreens.expand((Rect screen) {
+        final List<Rect> results = <Rect>[];
+        if (screen.top >= bounds.top && screen.bottom <= bounds.bottom) {
+          // Display feature splits the screen vertically
+          if (screen.left < bounds.left) {
+            // There is a smaller sub-screen, left of the display feature
+            results.add(Rect.fromLTWH(screen.left, screen.top,
+                bounds.left - screen.left, screen.height));
+          }
+          if (screen.right > bounds.right) {
+            // There is a smaller sub-screen, right of the display feature
+            results.add(Rect.fromLTWH(bounds.right, screen.top,
+                screen.right - bounds.right, screen.height));
+          }
+        } else if (screen.left >= bounds.left && screen.right <= bounds.right) {
+          // Display feature splits the sub-screen horizontally
+          if (screen.top < bounds.top) {
+            // There is a smaller sub-screen, above the display feature
+            results.add(Rect.fromLTWH(
+                screen.left, screen.top, screen.width, bounds.top - screen.top));
+          }
+          if (screen.bottom > bounds.bottom) {
+            // There is a smaller sub-screen, below the display feature
+            results.add(Rect.fromLTWH(screen.left, bounds.bottom, screen.width,
+                screen.bottom - bounds.bottom));
+          }
+        } else {
+          results.add(screen);
+        }
+        return results;
+      });
+    }
+    return subScreens;
+  }
+
+  static Offset _finiteOffset(Offset offset) {
+    if (offset.isFinite) {
+      return offset;
+    } else {
+      return Offset(_finiteNumber(offset.dx), _finiteNumber(offset.dy));
+    }
+  }
+
+  static double _finiteNumber(double nr) {
+    if (!nr.isInfinite) {
```

Comment:
should this just be "if (nr.isFinite) {"?

Reply:
Replaced this completely to **cap the anchorPoint to the size of the parent**, rather than maxFinite. Caping to maxFinite gave bad results when calculating distances, due to precision loss and infinity contagion in some cases. It was a bad idea on my part.